packages/backend/src/services/ValidationService/ValidationService.ts

```diff
@@ -0,0 +1,262 @@
+                !chart.metricQuery.tableCalculations
+                    ?.map((tableCalculation) => tableCalculation.name)
+                    ?.includes(metric);
+
+            const hasAccess = true; // TODO: hasSpaceAccess(chartuuid)
+            const commonValidation = {
+                createdAt: new Date(),
+                name: hasAccess ? chart.name : 'Private content',
+                chartUuid: chart.uuid,
+                table: chart.tableName,
+                projectUuid: chart.projectUuid,
+                lastUpdatedBy: `${chart.updatedByUser?.firstName} ${chart.updatedByUser?.lastName}`,
+            };
+            const dimensionErrors = chart.metricQuery.dimensions.reduce<
+                ValidationResponse[]
+            >((acc, field) => {
+                if (!existingDimensionIds?.includes(field)) {
+                    return [
+                        ...acc,
+                        {
+                            ...commonValidation,
+                            summary: `dimension not found ${field}`,
+                            error: `dimension not found ${field}`,
+                        },
+                    ];
+                }
+                return acc;
+            }, []);
+            const metricErrors = chart.metricQuery.metrics
+                .filter(filterAdditionalMetrics)
+                .reduce<ValidationResponse[]>((acc, field) => {
+                    if (!existingMetricIds?.includes(field)) {
+                        return [
+                            ...acc,
+                            {
+                                ...commonValidation,
+                                summary: `metric not found ${field}`,
+                                error: `metric not found ${field}`,
+                            },
+                        ];
+                    }
+                    return acc;
+                }, []);
+
+            const filterErrors = getFilterRules(
+                chart.metricQuery.filters,
+            ).reduce<ValidationResponse[]>((acc, field) => {
+                if (!existingFieldIds?.includes(field.target.fieldId)) {
+                    return [
+                        ...acc,
+                        {
+                            ...commonValidation,
+                            summary: `filter not found ${field}`,
+                            error: `filter not found ${field}`,
+                        },
+                    ];
+                }
+                return acc;
+            }, []);
+
+            const sortErrors = chart.metricQuery.sorts.reduce<
+                ValidationResponse[]
+            >((acc, field) => {
+                if (!existingFieldIds?.includes(field.fieldId)) {
+                    return [
+                        ...acc,
+                        {
+                            ...commonValidation,
+                            summary: `sort not found ${field}`,
```

Comment:
this `field` value is an object, so we have to get one of the key values or something 🤔

Reply:
all good now 🙌